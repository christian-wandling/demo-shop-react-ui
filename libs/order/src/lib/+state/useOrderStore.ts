import { create } from 'zustand';
import { getApi, OrderResponse, OrderStatus } from '@demo-shop-react-ui/api';

interface OrderState {
  orders: OrderResponse[];

  getSortedOrders: () => OrderResponse[];
  getOrderById: (id: number) => OrderResponse | undefined;
  fetchOrders: () => Promise<void>;
  fetchOrderById: (id: number) => Promise<void>;
}

export const useOrderStore = create<OrderState>()((set, get) => {
  return {
    orders: [],

    fetchOrders: async () => {
      const { orderApi } = getApi();
      const res = await orderApi.getAllOrdersOfCurrentUser();
      set({ orders: res.items });
    },

    fetchOrderById: async (id: number) => {
      const { orderApi } = getApi();
      const order = await orderApi.getOrderById({ id });
      set(state => ({ orders: [...state.orders, order] }));
    },

    getSortedOrders: () => {
      const { orders } = get();

      return orders.sort((a, b) => {
        if (a.status === b.status) {
          return a.created > b.created ? -1 : 1;
        }

        return a.status === OrderStatus.Created ? -1 : 1;
      });
    },

    getOrderById: id => {
      return get().orders.find(order => order.id === id);
    },
  };
});
