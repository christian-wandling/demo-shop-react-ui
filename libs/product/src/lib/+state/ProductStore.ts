import { create } from 'zustand';
import { Configuration, ProductApi, ProductResponse } from '@demo-shop-react-ui/api';
import { AllowedProductFilterTypes, ProductFilter } from '../models/product-filter';

interface ProductState {
  products: ProductResponse[];
  filter: ProductFilter;

  setFilter: (filter: ProductFilter) => void;
  getFilteredProducts: () => ProductResponse[];
  getProductById: (id: number) => ProductResponse | undefined;
  fetchProducts: () => Promise<void>;
  fetchProductById: (id: number) => Promise<void>;
}

const api = new ProductApi(new Configuration({ basePath: '' }));

export const useProductStore = create<ProductState>()((set, get) => ({
  products: [],
  filter: {},

  fetchProducts: async () => {
    const res = await api.getAllProducts();
    set({ products: res.items });
  },

  fetchProductById: async (id: number) => {
    const product = await api.getProductById({ id });
    set(state => ({ products: [...state.products, product] }));
  },

  getFilteredProducts: () => {
    const { filter, products } = get();
    const filterValues = Object.entries(filter);

    if (filterValues.length === 0) {
      return products;
    }

    return products.filter(entity =>
      filterValues.every(([key, filterValue]) => {
        const value = entity[key as AllowedProductFilterTypes];

        if (Array.isArray(value)) {
          return value.some(item => item.toLowerCase().includes(filterValue.toLowerCase()));
        }

        return value.toLowerCase().includes(filterValue.toLowerCase());
      })
    );
  },

  getProductById: id => {
    return get().products.find(product => product.id === id);
  },

  setFilter: (filter: ProductFilter) => {
    set(state => ({ ...state, filter }));
  },
}));
