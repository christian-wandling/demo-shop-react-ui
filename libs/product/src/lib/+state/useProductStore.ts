import { create } from 'zustand';
import { getApi, ProductResponse } from '@demo-shop-react-ui/api';
import { AllowedProductFilterTypes, ProductFilter } from '../models/productFilter';

interface ProductState {
  products: ProductResponse[];
  filter: ProductFilter;

  setFilter: (filter: ProductFilter) => void;
  getFilteredProducts: () => ProductResponse[];
  getProductById: (id: number) => ProductResponse | undefined;
  fetchProducts: () => Promise<void>;
  fetchProductById: (id: number) => Promise<void>;
}

export const useProductStore = create<ProductState>()((set, get) => {
  return {
    products: [],
    filter: {
      name: '',
    },

    fetchProducts: async () => {
      const { productApi } = getApi();
      const res = await productApi.getAllProducts();
      set({ products: res.items });
    },

    fetchProductById: async (id: number) => {
      const { productApi } = getApi();
      const product = await productApi.getProductById({ id });
      set(state => ({ products: [...state.products, product] }));
    },

    getFilteredProducts: () => {
      const { filter, products } = get();
      const filterValues = Object.entries(filter);

      if (filterValues.length === 0) {
        return products;
      }

      return products.filter(entity =>
        filterValues.every(([key, filterValue]) => {
          const value = entity[key as AllowedProductFilterTypes];

          if (Array.isArray(value)) {
            return value.some(item => item.toLowerCase().includes(filterValue.toLowerCase()));
          }

          return value.toLowerCase().includes(filterValue.toLowerCase());
        })
      );
    },

    getProductById: id => {
      return get().products.find(product => product.id === id);
    },

    setFilter: (filter: ProductFilter) => {
      set(state => ({ ...state, filter }));
    },
  };
});
