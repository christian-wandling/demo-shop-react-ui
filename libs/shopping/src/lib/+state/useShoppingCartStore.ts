import { create } from 'zustand';
import { AddCartItemRequest, CartItemResponse, getApi, UpdateCartItemQuantityRequest } from '@demo-shop-react-ui/api';

interface ShoppingCartState {
  items: CartItemResponse[];
  sessionId: number | null;
  showCart: boolean;
  fetchCurrentSession: () => Promise<void>;
  add: (addCartItemRequest: AddCartItemRequest) => Promise<void>;
  remove: (id: number) => Promise<void>;
  updateItemQuantity: (id: number, updateCartItemQuantityRequest: UpdateCartItemQuantityRequest) => Promise<void>;
  getItemById: (id: number) => CartItemResponse | undefined;
  getItemByProductId: (productId: number) => CartItemResponse | undefined;
  setShowCart: (showCart: boolean) => void;

  getTotalPrice: () => number;

  getItemCount: () => number;

  hasActiveSession: () => boolean;
}

export const useShoppingCartStore = create<ShoppingCartState>()((set, get) => {
  return {
    items: [],
    sessionId: null,
    showCart: false,

    getTotalPrice(): number {
      return get().items.reduce((sum, item) => sum + item.totalPrice, 0);
    },

    getItemCount(): number {
      return get().items.reduce((sum, item) => sum + item.quantity, 0);
    },

    hasActiveSession(): boolean {
      return get().sessionId !== null;
    },

    fetchCurrentSession: async () => {
      const { shoppingSessionApi } = getApi();
      const session = await shoppingSessionApi.resolveCurrentShoppingSession();
      set({ sessionId: session.id, items: session.items });
    },

    add: async (addCartItemRequest: AddCartItemRequest) => {
      if (!get().hasActiveSession()) {
        throw new Error('No active shopping session');
      }

      const itemInCart = get().getItemByProductId(addCartItemRequest.productId);

      if (itemInCart) {
        get().updateItemQuantity(itemInCart.id, { quantity: itemInCart.quantity + 1 });
        return;
      }

      const { shoppingSessionApi } = getApi();
      await shoppingSessionApi.addCartItem({ addCartItemRequest });
      get().fetchCurrentSession();
    },

    remove: async (id: number) => {
      if (!get().hasActiveSession()) {
        throw new Error('No active shopping session');
      }

      const itemInCart = get().getItemById(id);

      if (!itemInCart) {
        throw new Error('Item not found');
      }

      if (itemInCart.quantity > 1) {
        get().updateItemQuantity(itemInCart.id, { quantity: itemInCart.quantity - 1 });
        return;
      }

      const { shoppingSessionApi } = getApi();
      await shoppingSessionApi.removeCartItem({ id });
      get().fetchCurrentSession();
    },

    updateItemQuantity: async (id: number, updateCartItemQuantityRequest: UpdateCartItemQuantityRequest) => {
      if (!get().hasActiveSession()) {
        throw new Error('No active shopping session');
      }

      const { shoppingSessionApi } = getApi();
      await shoppingSessionApi.updateCartItemQuantity({ id, updateCartItemQuantityRequest });
      get().fetchCurrentSession();
    },

    getItemById: id => {
      return get().items.find(item => item.id === id);
    },

    getItemByProductId: productId => {
      return get().items.find(item => item.productId === productId);
    },

    setShowCart: (showCart: boolean) => {
      set(state => ({ ...state, showCart }));
    },
  };
});
