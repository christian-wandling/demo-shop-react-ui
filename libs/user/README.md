# @demo-shop-react-ui/user

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test @demo-shop-react-ui/user` to execute the unit tests via [Vitest](https://vitest.dev/).
