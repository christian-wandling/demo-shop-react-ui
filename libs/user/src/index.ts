export * from './lib/+state/useUserStore';
