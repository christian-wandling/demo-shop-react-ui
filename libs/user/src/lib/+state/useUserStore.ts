import { create } from 'zustand';
import { getApi, UpdateUserAddressRequest, UpdateUserPhoneRequest, UserResponse } from '@demo-shop-react-ui/api';

interface UserState {
  user: UserResponse | null;

  fetchCurrentUser: () => Promise<void>;
  updateUserAddress: (address: UpdateUserAddressRequest) => Promise<void>;
  updateUserPhone: (address: UpdateUserPhoneRequest) => Promise<void>;
}

export const useUserStore = create<UserState>()((set, get) => {
  return {
    user: null,

    fetchCurrentUser: async () => {
      const { userApi } = getApi();
      const user = await userApi.resolveCurrentUser();
      set({ user });
    },

    updateUserAddress: async (updateUserAddressRequest: UpdateUserAddressRequest) => {
      const { userApi } = getApi();
      const address = await userApi.updateCurrentUserAddress({ updateUserAddressRequest });
      set(state => ({ user: state.user && { ...state.user, address } }));
    },

    updateUserPhone: async (updateUserPhoneRequest: UpdateUserPhoneRequest) => {
      const { userApi } = getApi();
      const user = await userApi.updateCurrentUserPhone({ updateUserPhoneRequest });
      set({ user });
    },
  };
});
