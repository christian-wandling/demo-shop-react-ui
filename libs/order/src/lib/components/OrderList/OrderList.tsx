import { useEffect } from 'react';
import { Link, useNavigate } from 'react-router';
import { useUserStore } from '@demo-shop-react-ui/user';
import { DateTime, formatCurrency } from '@demo-shop-react-ui/shared';
import { useOrderStore } from '../../+state/useOrderStore';
import { OrderStatusBadge } from '../OrderStatusBadge/OrderStatusBadge';

/**
 * Component that displays a list of orders for the current user.
 * The orders are sorted by status and date.
 */
export default function OrderList() {
  const MAX_THUMBNAILS = 5;

  useOrderStore(state => state.orders);
  const user = useUserStore(state => state.user);
  const getSortedOrders = useOrderStore(state => state.getSortedOrders);
  const fetchOrders = useOrderStore(state => state.fetchOrders);
  const navigate = useNavigate();

  useEffect(() => {
    fetchOrders();
  }, [fetchOrders]);

  return (
    <div className="mx-auto px-8 max-w-7xl" data-testid="order-list">
      <section className="py-8 top-16 sticky z-10 bg-white border-b-2 border-gray-100">
        {user && (
          <>
            <h1 className="font-bold text-gray-700" data-testid="order-list-username">
              {`${user.firstname} ${user.lastname}`}
            </h1>
            <div className="text-xs text-gray-500">
              <p data-testid="order-list-address1">{`${user.address?.street} ${user.address?.apartment}`}</p>
              <p data-testid="order-list-address2">{`${user.address?.zip} ${user.address?.city}, ${user.address?.region}`}</p>
              <p data-testid="order-list-address3">{user.address?.country}</p>
            </div>
          </>
        )}
      </section>
      <section>
        <ul className="divide-y divide-gray-100">
          {getSortedOrders().map(order => (
            <li
              key={order.id}
              onClick={() => navigate(`./${order.id}`)}
              className="relative flex flex-col md:flex-row justify-between gap-x-3 sm:gap-x-6 py-2 sm:py-5 sm:px-5 cursor-pointer hover:bg-slate-50"
              data-testid={`order-list-item-link${order.id}`}>
              <div className="flex min-w-0 gap-x-3 mb-4">
                <div className="min-w-0 flex-auto">
                  <p
                    className="text-xl mb-4 font-semibold hover:underline"
                    data-testid={`order-list-item-id${order.id}`}>
                    Order #{order.id}
                  </p>
                  <div className="flex gap-1 sm:gap-4">
                    {order.items.slice(0, MAX_THUMBNAILS).map((item, index) => {
                      if (index === MAX_THUMBNAILS - 1 && order.items.length > MAX_THUMBNAILS) {
                        return (
                          <div
                            key={item.productId}
                            className="w-12 h-12 sm:w-24 sm:h-24 aspect-1 flex-shrink-0 bg-gray-800 text-xl sm:text-5xl text-white flex items-center justify-center rounded-lg"
                            data-testid={`order-list-item-thumbnails-more${order.id}`}>
                            +{order.items.length - MAX_THUMBNAILS + 1}
                          </div>
                        );
                      } else {
                        return (
                          <Link
                            key={item.productId}
                            to={`/products/${item.productId}`}
                            onClick={e => e.stopPropagation()}
                            className="relative w-12 h-12 sm:w-24 sm:h-24 aspect-1 flex-shrink-0 rounded-lg overflow-hidden"
                            data-testid={`order-list-item-product-link${order.id}-${item.productId}`}>
                            <img
                              src={item.productThumbnail}
                              alt={item.productName}
                              title={item.productName}
                              className="w-full h-full object-cover hover:opacity-75"
                              data-testid={`order-list-item-thumbnails${order.id}-${item.productId}`}
                            />
                          </Link>
                        );
                      }
                    })}
                  </div>
                </div>
              </div>
              <OrderStatusBadge
                status={order.status}
                className="sm:absolute sm:right-5 sm:top-5 mb-4"
                data-testid={`order-list-item-status${order.id}`}
              />
              <div className="flex flex-col md:items-end sm:justify-end gap-1">
                <p className="text-sm leading-none text-gray-900" data-testid={`order-list-item-amount${order.id}`}>
                  {formatCurrency(order.amount)}
                </p>
                <DateTime
                  dateTime={order.created}
                  className="text-xs leading-none text-gray-500"
                  data-testid={`order-list-item-date${order.id}`}
                />
              </div>
            </li>
          ))}
        </ul>
      </section>
    </div>
  );
}
