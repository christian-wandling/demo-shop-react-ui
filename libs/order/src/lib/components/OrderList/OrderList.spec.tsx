import { beforeEach, describe, expect, it, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import OrderList from './OrderList';
import { useUserStore } from '@demo-shop-react-ui/user';
import { useOrderStore } from '../../+state/useOrderStore';
import { useNavigate } from 'react-router';

vi.mock('react-router', () => ({
  Link: ({ children, ...props }: any) => <a {...props}>{children}</a>,
  useNavigate: vi.fn(),
}));

vi.mock('@demo-shop-react-ui/user', () => ({
  useUserStore: vi.fn(),
}));

vi.mock('../../+state/useOrderStore', () => ({
  useOrderStore: vi.fn(),
}));

vi.mock('@demo-shop-react-ui/shared', async importOriginal => ({
  ...(await importOriginal()),
  DateTime: ({ dateTime, className }: any) => <div className={className}>{dateTime}</div>,
}));

describe('OrderList', () => {
  const mockUser = {
    firstname: 'John',
    lastname: 'Doe',
    address: {
      street: '123 Main St',
      apartment: 'Apt 4B',
      zip: '12345',
      city: 'Springfield',
      region: 'IL',
      country: 'USA',
    },
  };

  const mockOrders = [
    {
      id: '1001',
      status: 'delivered',
      amount: 249.99,
      created: '2023-06-15T10:30:00Z',
      items: [
        {
          productId: 'p101',
          productName: 'Test Product 1',
          productThumbnail: '/test1.jpg',
        },
        {
          productId: 'p102',
          productName: 'Test Product 2',
          productThumbnail: '/test2.jpg',
        },
      ],
    },
    {
      id: '1002',
      status: 'processing',
      amount: 99.5,
      created: '2023-06-20T14:20:00Z',
      items: [
        {
          productId: 'p103',
          productName: 'Test Product 3',
          productThumbnail: '/test3.jpg',
        },
      ],
    },
  ];

  const mockNavigate = vi.fn();
  const mockFetchOrders = vi.fn();
  const mockGetSortedOrders = vi.fn().mockReturnValue(mockOrders);

  beforeEach(() => {
    vi.clearAllMocks();
    vi.mocked(useNavigate).mockReturnValue(mockNavigate);
    vi.mocked(useUserStore).mockReturnValue(mockUser);
    vi.mocked(useOrderStore).mockImplementation(selector => {
      if (selector.toString().includes('fetchOrders')) return mockFetchOrders;
      if (selector.toString().includes('getSortedOrders')) return mockGetSortedOrders;
      return mockOrders;
    });
  });

  it('fetches orders on mount', () => {
    render(<OrderList />);
    expect(mockFetchOrders).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const { baseElement } = render(<OrderList />);

    expect(baseElement).toMatchSnapshot();
  });

  it('displays user information correctly', () => {
    render(<OrderList />);
    expect(screen.getByText('John Doe')).toBeTruthy();
    expect(screen.getByText('123 Main St Apt 4B')).toBeTruthy();
    expect(screen.getByText('12345 Springfield, IL')).toBeTruthy();
    expect(screen.getByText('USA')).toBeTruthy();
  });

  it('displays order list with correct number of orders', () => {
    render(<OrderList />);
    const orderElements = screen.getAllByText(/Order #/);
    expect(orderElements).toHaveLength(2);
    expect(screen.getByText('Order #1001')).toBeTruthy();
    expect(screen.getByText('Order #1002')).toBeTruthy();
  });

  it('displays correct order amount', () => {
    render(<OrderList />);
    expect(screen.getByText('$249.99')).toBeTruthy();
    expect(screen.getByText('$99.50')).toBeTruthy();
  });

  it('navigates to order details when clicking an order', () => {
    render(<OrderList />);
    const firstOrder: HTMLLIElement = screen.getByText('Order #1001').closest('li')!;

    fireEvent.click(firstOrder);

    expect(mockNavigate).toHaveBeenCalledWith('./1001');
  });

  it('limits product thumbnails to MAX_THUMBNAILS', () => {
    const manyItemsOrder = {
      id: '1003',
      status: 'delivered',
      amount: 599.99,
      created: '2023-06-25T09:15:00Z',
      items: Array.from({ length: 8 }).map((_, i) => ({
        productId: `p${200 + i}`,
        productName: `Test Product ${200 + i}`,
        productThumbnail: `/test${200 + i}.jpg`,
      })),
    };

    mockGetSortedOrders.mockReturnValue([manyItemsOrder]);

    render(<OrderList />);

    expect(screen.getByText('+4')).toBeTruthy();
  });
});
