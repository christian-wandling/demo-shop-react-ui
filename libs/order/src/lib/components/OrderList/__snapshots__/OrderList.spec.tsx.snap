// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OrderList > matches snapshot 1`] = `
<body>
  <div>
    <div
      class="mx-auto px-8 max-w-7xl"
    >
      <section
        class="py-8 top-16 sticky z-10 bg-white border-b-2 border-gray-100"
      >
        <h1
          class="font-bold text-gray-700"
        >
          John Doe
        </h1>
        <div
          class="text-xs text-gray-500"
        >
          <p>
            123 Main St Apt 4B
          </p>
          <p>
            12345 Springfield, IL
          </p>
          <p>
            USA
          </p>
        </div>
      </section>
      <section>
        <ul
          class="divide-y divide-gray-100"
        >
          <li
            class="relative flex flex-col md:flex-row justify-between gap-x-3 sm:gap-x-6 py-2 sm:py-5 sm:px-5 cursor-pointer hover:bg-slate-50"
          >
            <div
              class="flex min-w-0 gap-x-3 mb-4"
            >
              <div
                class="min-w-0 flex-auto"
              >
                <p
                  class="text-xl mb-4 font-semibold hover:underline"
                >
                  Order #
                  1001
                </p>
                <div
                  class="flex gap-1 sm:gap-4"
                >
                  <a
                    class="relative w-12 h-12 sm:w-24 sm:h-24 aspect-1 flex-shrink-0 rounded-lg overflow-hidden"
                    to="/products/p101"
                  >
                    <img
                      alt="Test Product 1"
                      class="w-full h-full object-cover hover:opacity-75"
                      src="/test1.jpg"
                      title="Test Product 1"
                    />
                  </a>
                  <a
                    class="relative w-12 h-12 sm:w-24 sm:h-24 aspect-1 flex-shrink-0 rounded-lg overflow-hidden"
                    to="/products/p102"
                  >
                    <img
                      alt="Test Product 2"
                      class="w-full h-full object-cover hover:opacity-75"
                      src="/test2.jpg"
                      title="Test Product 2"
                    />
                  </a>
                </div>
              </div>
            </div>
            <div
              class="sm:absolute sm:right-5 sm:top-5 mb-4 rounded-lg px-2 sm:px-4 py-1 sm:py-1.5 max-w-20 sm:max-w-28 text-center text-white border border-slate-300 text-xs "
              data-testid="orderStatus"
            >
              Delivered
            </div>
            <div
              class="flex flex-col md:items-end sm:justify-end gap-1"
            >
              <p
                class="text-sm leading-none text-gray-900"
              >
                $249.99
              </p>
              <div
                class="text-xs leading-none text-gray-500"
              >
                2023-06-15T10:30:00Z
              </div>
            </div>
          </li>
          <li
            class="relative flex flex-col md:flex-row justify-between gap-x-3 sm:gap-x-6 py-2 sm:py-5 sm:px-5 cursor-pointer hover:bg-slate-50"
          >
            <div
              class="flex min-w-0 gap-x-3 mb-4"
            >
              <div
                class="min-w-0 flex-auto"
              >
                <p
                  class="text-xl mb-4 font-semibold hover:underline"
                >
                  Order #
                  1002
                </p>
                <div
                  class="flex gap-1 sm:gap-4"
                >
                  <a
                    class="relative w-12 h-12 sm:w-24 sm:h-24 aspect-1 flex-shrink-0 rounded-lg overflow-hidden"
                    to="/products/p103"
                  >
                    <img
                      alt="Test Product 3"
                      class="w-full h-full object-cover hover:opacity-75"
                      src="/test3.jpg"
                      title="Test Product 3"
                    />
                  </a>
                </div>
              </div>
            </div>
            <div
              class="sm:absolute sm:right-5 sm:top-5 mb-4 rounded-lg px-2 sm:px-4 py-1 sm:py-1.5 max-w-20 sm:max-w-28 text-center text-white border border-slate-300 text-xs "
              data-testid="orderStatus"
            >
              Processing
            </div>
            <div
              class="flex flex-col md:items-end sm:justify-end gap-1"
            >
              <p
                class="text-sm leading-none text-gray-900"
              >
                $99.50
              </p>
              <div
                class="text-xs leading-none text-gray-500"
              >
                2023-06-20T14:20:00Z
              </div>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </div>
</body>
`;
